### PR TITLE
sonar fix

### PR DESCRIPTION
**Pull request by sealDebolina**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the enable_keep_helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/common/summary/utils.ts [hunk 1]
```diff
@@ -196,16 +196,20 @@ export const proceedingSummaryData = (
...
+        : applicantOrderDetails(userCase, courtOrderDetails),
       changeUrl: isRespondent ? PROCEEDINGS_COURT_PROCEEDINGS : C100_OTHER_PROCEEDINGS_DETAILS,
     },
     ...(isRespondent
       ? OPotherProceedingsSessionParserUtilRespondent(userCase, keys, 'courtProceedingsOrders', language)
       : OPotherProceedingsSessionParserUtil(userCase, keys, 'op_courtProceedingsOrders', language)),
   ];
 };
+
+const applicantOrderDetails = (userCase: Partial<CaseWithId>, courtOrderDetails: string) => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/summary/utils.ts:209]: mention return type

File: src/main/steps/common/summary/utils.ts [hunk 2]
```diff
@@ -196,16 +196,20 @@ export const proceedingSummaryData = (
...
+  return userCase.hasOwnProperty('op_courtProceedingsOrders') ? courtOrderDetails?.split(',').join('') : '';
+};
+
+const respondentOrderDetails = (userCase: Partial<CaseWithId>, courtOrderDetails: string) => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/summary/utils.ts:213]: mention return type

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 3]
```diff
@@ -202,27 +234,31 @@ export const prepareRequest = (caseData: Partial<Case>): string => {
...
-  }, {});
-
-  return JSON.stringify(transformedCaseData.respondingCitizenAoH, function (k, v) {
-    return v === undefined ? null : v;
-  });
 };
-const updateCaseDataMapper = {
-  c1A: 'respondingCitizenAoH',
+
+const dataCleanupForNoSafetyConcern = (c1A_haveSafetyConcerns: YesOrNo | undefined, data) => {
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:239]: mention return type

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 4]
```diff
@@ -173,7 +150,62 @@ export const prepareRequest = (caseData: Partial<Case>): string => {
...
+    c1A_safetyConernAbout?.includes(C1ASafteyConcernsAbout.CHILDREN) &&
+    !c1A_concernAboutChild?.includes(C1AAbuseTypes.SEXUAL_ABUSE)
+  ) {
+    if (data.c1A_safteyConcerns?.child?.sexualAbuse) {
+      data.c1A_safteyConcerns.child.sexualAbuse = undefined;
+    }
+  }
+};
+
+const dataCleanUpForAbduction = (
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:202]: mention return type

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 5]
```diff
@@ -173,7 +150,62 @@ export const prepareRequest = (caseData: Partial<Case>): string => {
       data.c1A_safteyConcerns.respondent.sexualAbuse = undefined;
     }
   }
+};
 
+const dataCleanUpForChildAbuse = (
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:155]: mention return type

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 6]
```diff
@@ -93,46 +71,45 @@ export const prepareRequest = (caseData: Partial<Case>): string => {
...
+
+  return JSON.stringify(transformedCaseData.respondingCitizenAoH, function (k, v) {
+    return v === undefined ? null : v;
+  });
+};
+const updateCaseDataMapper = {
+  c1A: 'respondingCitizenAoH',
+};
+
+const dataCleanUpForRespondentAbuse = (
```
Comment (vivek-sekhar) [inline src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:108]: mention return type

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts [hunk 7]
```diff
@@ -93,46 +71,45 @@ export const prepareRequest = (caseData: Partial<Case>): string => {
...
-  }
+
+    return transformedData;
+  }, {});
+
+  return JSON.stringify(transformedCaseData.respondingCitizenAoH, function (k, v) {
+    return v === undefined ? null : v;
+  });
+};
+const updateCaseDataMapper = {
```
[the remark this answers is not in the document]
Comment (sealDebolina) [reply src/main/steps/common/safety-concerns/review/safetyConcernMapper.ts:104]: needed

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**